core/src/main/scala/org/apache/pekko/persistence/jdbc/query/javadsl/JdbcReadJournal.scala

```diff
@@ -132,4 +138,16 @@ class JdbcReadJournal(journal: ScalaJdbcReadJournal)
    */
   override def eventsByTag(tag: String, offset: Offset): Source[EventEnvelope, NotUsed] =
     journal.eventsByTag(tag, offset).asJava
+
+  /**
+   * Returns the last known sequence number for the given `persistenceId`. Empty if the `persistenceId` is unknown.
+   *
+   * @param persistenceId The `persistenceId` for which the last known sequence number should be returned.
+   * @return Some sequence number or None if the `persistenceId` is unknown.
+   */
+  def currentLastKnownSequenceNumberByPersistenceId(persistenceId: String): CompletionStage[Optional[Long]] =
```

Comment:
I wonder if this should be java.lang.Long instead of scala.Long? scala.Long acts as an alias for the Java long primitive which is not the same as java.lang.Long. java.lang.Long is nullable but I still prefer the Optional wrapper.

wdyt @janjaali @ptrdom ?

Reply:
I was considering `scala.Long` as the better suited one here, since the `EventEnvelope` itself - which is the return type of multiple other methods in this class - carries the offset and specially the sequenceNr - the attribute that this method is actually interested in - as scala.Long.

The Optional wrapper was "inspired" by the `GetObjectResult` for the `DurableStateStore` javadsl.

Reply:
Quick search in Pekko repository gives mixed results, some `javadsl` classes stick to `java.lang.Long` quite well, others do not. My guess is that `javadsl` should use `java.lang.Long`, it is just difficult to enforce, because there is no automatic linting for it.

Reply:
I agree that Pekko is inconsistent but I would prefer that we use java.lang.Long here.

Reply:
Using `java.lang.Long` in <https://github.com/apache/pekko-persistence-jdbc/pull/267/commits/711145fb39f9a71dbf8455fa94af4adcfa7a049d>. I wasn't able to import `java.lang.Long` in top level since `scala.Long` already used in other query methods.